File: leetcode/1614-maximum-nesting-depth-of-the-parentheses.js
// https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
  let _maxDepth = 0;
  let depth = 0;
  for (let c of s) {
    if (c === '(') depth += 1;
    else if (c === ')') depth -= 1;
    _maxDepth = Math.max(_maxDepth, depth);
  }

  return _maxDepth;
};

File: leetcode/1614-maximum-nesting-depth-of-the-parentheses.py
# https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

class Solution:
    def maxDepth(self, s: str) -> int:
        max_depth = 0
        cur_depth = 0
        for c in s:
            if c == '(':
              cur_depth += 1
            elif c == ')':
              cur_depth -= 1
            max_depth = max(max_depth, cur_depth)
        
        return max_depth
